### PR TITLE
MathSAT Driver

### PR DESCRIPTION
**Pull request by mfarif**
MathSAT driver integration.

### CODE REVIEW

File: src/flags.ml [hunk 1]
```diff
@@ -3266,6 +3293,21 @@ let solver_dependant_actions () =
       )
     | None -> Log.log L_warn "Couldn't determine Boolector version"
   )
+  | `MathSAT_SMTLIB -> (
+    let cmd = Format.asprintf "%s -version" (Smt.mathsat_bin ()) in
+    match get_version true cmd with
+    | Some (major_rev, minor_rev, patch_rev) ->
+      if major_rev < 5 || (major_rev = 5 && (minor_rev < 5 || patch_rev < 4))  then (
```
Comment (daniel-larraz) [inline src/flags.ml:3300]: The right condition is:
`if major_rev < 5 || (major_rev = 5 && (minor_rev < 5 || (minor_rev = 5 && patch_rev < 4)))  then (`

File: src/smt/MathSATDriver.ml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  match l with
+  | `Inferred fs ->
+        GenericSMTLIBDriver.string_of_logic (`Inferred fs)
+  | `None -> "ALL"
+  | `SMTLogic s ->
+      if String.contains s 'S' then
+        failwith "MathSAT do not support Theory of Strings"
+      else s
+
+let pp_print_logic fmt l = Format.pp_print_string fmt (string_of_logic l)
```
Comment (daniel-larraz) [inline src/smt/MathSATDriver.ml:80]: Remove all these lines. They are not necessary.

File: src/smt/MathSATDriver.ml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    preprocessor.simplification=2
+    theory.bv.eager=true
+   *)
+
+  let cmd = [| mathsat_bin |] in
+    match timeout with
+    | None -> cmd
+    | Some timeout ->
+      let timeout =
+        Format.sprintf "--tlimit=%.0f" ((1000.0 *. timeout) |> ceil)
```
Comment (daniel-larraz) [inline src/smt/MathSATDriver.ml:59]: The right MathSAT timeout option is` "-timeout=%.0f"`, which is only supported in the current nightly build.

Comment (mfarif) [reply src/smt/MathSATDriver.ml]: Corrected.

File: src/smt/MathSATDriver.ml (hunk 2 above)
Comment (mfarif) [reply src/smt/MathSATDriver.ml]: Removed. Thank you for pointing out.

File: src/flags.ml (hunk 1 above)
Comment (mfarif) [reply src/flags.ml]: corrected. Thanks for pointing out.

**Pull request merged**